sdk/bcs/src/index.ts

```diff
@@ -371,17 +371,36 @@ function ulebDecode(
  * BCS value or a part of a composed structure/vector.
  */
 export interface TypeInterface {
-  encode: (data: any, size: number) => BcsWriter;
-  decode: (data: Uint8Array) => any;
+  encode: (data: any, size: number, typeParams: string[]) => BcsWriter;
+  decode: (data: Uint8Array, typeParams: string[]) => any;
 
-  _encodeRaw: (writer: BcsWriter, data: any) => BcsWriter;
-  _decodeRaw: (reader: BcsReader) => any;
+  _encodeRaw: (writer: BcsWriter, data: any, typeParams: string[]) => BcsWriter;
+  _decodeRaw: (reader: BcsReader, typeParams: string[]) => any;
 }
 
+/**
+ *
+ */
+export type TypeDefinition = {
+
+};
+
+/**
+ *
+ */
+export type TypeSchema = {
```

Comment:
Is `Schema` the established name for what this is?  I imagine that it's quite an overloaded word in the BCS world, e.g. when I read `TypeSchema` I understand it to be similar to what `TypeInterface` is right now.  The content of this struct suggests to me a name like `BCSConfig` or maybe `TypeGrammar`?

Reply:
I was planning to have two groups of type definitions:
```json
{
  "structs": {
    "Coin<T>": {
      "value": "u64"
    }
   },
   "enums": {
     "Option<T>": {
       "none": null,
       "some": "T"
     }
   }
}
```